ethereum-consensus/src/capella/block_processing.rs

```diff
@@ -37,10 +41,33 @@ pub fn process_bls_to_execution_change<
         BYTES_PER_LOGS_BLOOM,
         MAX_EXTRA_DATA_BYTES,
     >,
-    _signed_address_change: &mut SignedBlsToExecutionChange,
-    _context: &Context,
+    signed_address_change: &mut SignedBlsToExecutionChange,
+    context: &Context,
 ) -> Result<()> {
-    unimplemented!()
+    let address_change = signed_address_change;
+
+    assert!(address_change.message.validator_index < state.validators.len());
+
+    let validator = &mut state.validators[address_change.message.validator_index];
+
+    assert!(validator.withdrawal_credentials.starts_with(&[BLS_WITHDRAWAL_PREFIX]));
+    assert!(
+        validator.withdrawal_credentials[1..]
+            == hash(address_change.message.from_bls_public_key.as_ref())[1..]
+    );
+
+    let domain =
+        compute_domain(BlsToExecutionChange, None, Some(state.genesis_validators_root), context)?;
+    let signing_root = compute_signing_root(address_change, domain)?;
+    let pk: &BlsPublicKey = &address_change.message.from_bls_public_key;
+    assert!(verify_signature(&pk, signing_root.as_ref(), &address_change.signature,).is_ok());
```

Comment:
double-ref here and extra comma
```suggestion
    assert!(verify_signature(pk, signing_root.as_ref(), &address_change.signature).is_ok());
```

Reply:
also note, if we compute the `domain`, there's a higher-level function to use: https://github.com/ralexstokes/ethereum-consensus/blob/61836a64464f311a40fd3038edf774f5c239e3ea/ethereum-consensus/src/signing.rs#L33

Reply:
pushed with this